src/sources/envoy_als/grpc.rs

```diff
@@ -33,68 +37,99 @@ impl AccessLogService for Service {
             tokio::select! {
                 _ = &mut shutdown => break,
                 stream_msg = in_stream.message() => {
-                    match stream_msg {
-                        Ok(omsg) => {
-                            match omsg {
-                                Some(msg) => {
-                                    let mut events = vec![];
+                    match self.handle_message(stream_msg, &mut stream_identifier).await {
+                        Ok(_) => {},
+                        Err(err) => {
```

Comment:
Instead of matching with an empty `Ok` case, you can use 

```Rust
if let Err(err) = self.handle_message(stream_msg, &mut stream_identifier).await {
// ...
}
```